src/Stratis.Features.Diagnostic/DiagnosticSettings.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Features.Diagnostic.PeerDiagnostic;
+
+namespace Stratis.Features.Diagnostic
+{
+    /// <summary>
+    /// Configuration related to the API interface.
+    /// </summary>
+    public class DiagnosticSettings
+    {
+        /// <summary>The default value for peers statistics collector.</summary>
+        public const bool DefaultPeersStatisticsCollectorEnabled = false;
+        
+        /// <summary>The default value for maximum peers logged events.</summary>
+        public const int DefaultMaxPeerLoggedEvents = 10;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether <see cref="PeerStatisticsCollector"/> starts enabled.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if <see cref="PeerStatisticsCollector"/> starts enabled; otherwise, <c>false</c>.
+        /// </value>
+        public bool PeersStatisticsCollectorEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum peer logged events in <see cref="PeerStatisticsCollector"/>.
+        /// </summary>
+        /// <value>
+        /// The maximum peer logged events.
+        /// </value>
+        public int MaxPeerLoggedEvents { get; set; }
+
+        /// <summary>
+        /// Initializes an instance of the object from the node configuration.
+        /// </summary>
+        /// <param name="nodeSettings">The node configuration.</param>
+        public DiagnosticSettings(NodeSettings nodeSettings)
+        {
+            Guard.NotNull(nodeSettings, nameof(nodeSettings));
+
+            this.logger = nodeSettings.LoggerFactory.CreateLogger(typeof(DiagnosticSettings).FullName);
```

Comment:
`this.logger` should be set to `this.GetType().FullName` ? Then its inline with how we do it in other places.

Reply:
yes agree, I copied this setting class from ApiSettings because it was the last setting created, so even that class should be changed